x/nexus/keeper/grpc_query.go

```diff
@@ -93,19 +109,27 @@ func (k Keeper) TransferFee(c context.Context, req *types.TransferFeeRequest) (*
 		return nil, err
 	}
 
-	if !k.IsAssetRegistered(ctx, sourceChain, amount.Denom) {
+	if !q.keeper.IsAssetRegistered(ctx, sourceChain, amount.Denom) {
 		return nil, sdkerrors.Wrapf(types.ErrNexus, "%s is not a registered asset on chain %s", amount.Denom, sourceChain.Name)
 	}
 
-	if !k.IsAssetRegistered(ctx, destinationChain, amount.Denom) {
+	if !q.keeper.IsAssetRegistered(ctx, destinationChain, amount.Denom) {
 		return nil, sdkerrors.Wrapf(types.ErrNexus, "%s is not a registered asset on chain %s", amount.Denom, destinationChain.Name)
 	}
 
 	if amount.IsNegative() {
 		return nil, fmt.Errorf("amount cannot be negative")
 	}
 
-	fee, err := k.ComputeTransferFee(ctx, sourceChain, destinationChain, amount)
+	// When source chain is another cosmos chain, use axelarnet for fee info where deposit address is generated on
+	if q.axelarnet.IsCosmosChain(ctx, sourceChain.Name) && !strings.EqualFold(sourceChain.Name, exported.Axelarnet.Name) {
+		sourceChain, ok = q.keeper.GetChain(ctx, exported.Axelarnet.Name)
+		if !ok {
+			return nil, sdkerrors.Wrapf(types.ErrNexus, "%s is not a registered chain", exported.Axelarnet.Name)
+		}
+	}
+
+	fee, err := q.keeper.ComputeTransferFee(ctx, sourceChain, destinationChain, amount)
```

Comment:
```suggestion
	// When source chain is another cosmos chain, use axelarnet for fee info where deposit address is generated on
	feeCalcSourceChain := sourceChain
	if q.axelarnet.IsCosmosChain(ctx, feeCalcSourceChain.Name) {
		feeCalcSourceChain = exported.Axelarnet
	}

	fee, err := q.keeper.ComputeTransferFee(ctx, feeCalcSourceChain, destinationChain, amount)
```

Reply:
It's cleaner this way and also by creating a new variable you make it clearer that it's intentional instead of a bug.